src/chk/chk_common.c

```diff
@@ -233,6 +237,14 @@ chk_pending_alloc(struct btr_instance *tins, d_iov_t *key_iov, d_iov_t *val_iov,
 	cpr->cpr_class = cpb->cpb_class;
 	cpr->cpr_action = CHK__CHECK_INCONSIST_ACTION__CIA_INTERACT;
 
+	D_ASSERTF(cpb->cpb_option_nr <= 3, "Too many options %u > 3\n", cpb->cpb_option_nr);
```

Comment:
Seems like maybe overly strict to assert on. Is there logic somewhere depending on this limit being true?

Reply:
It is CHK internal logic to generate the pending options, neither depends on user input nor the on-disk layout. Assertion check will guarantee that the CHK internal logic works as expectedly.